module4/Homework4Playground.playground/Contents.swift

```diff
@@ -0,0 +1,211 @@
+
+print("b) ---------------")
+
+//Call the function twice, sending a different Animal each time.
+
+enum Animal {
+  case dog
+  case cat
+  case fish
+  case bird
+}
+
+func theSoundMadeBy(animal: Animal) {
+  switch animal {
+  case .dog:
+    print("A dog goes woof")
+  case .cat:
+    print("A cat goes meow")
+  case .fish:
+    print("A fish goes bloop")
+  case .bird:
+    print("A bird goes tweet")
+  }
+}
+
+theSoundMadeBy(animal: .dog)
+theSoundMadeBy(animal: .cat)
+theSoundMadeBy(animal: .fish)
+theSoundMadeBy(animal: .bird)
+
+
+
+//c) This question will have you creating multiple functions that will require you to use closures and collections. First, you will do some setup.
+
+print("c) ---------------")
+
+//Create an array of Int called nums with the values of 0 to 100.
+
+let nums = Array(0...100)
+
+//Create an array of Int? called numsWithNil with the following values:
+//79, nil, 80, nil, 90, nil, 100, 72
+
+let numsWithNil: [Int?] = [79, nil, 80, nil, 90, nil, 100, 72]
+
+//Create an array of Int called numsBy2 with values starting at 2 through 100, by 2.
+
+var numsBy2: [Int] = Array(1...50).map { $0 * 2 }
+print(numsBy2)
+numsBy2 = Array(stride(from:2, through: 100, by: 2))
+print(numsBy2)
+
+//Create an array of Int called numsBy4 with values starting at 2 through 100, by 4.
+
+// Is this what is meant?
+var numsBy4: [Int] = Array(1...25).map { $0 * 4 - 2 }
+print(numsBy4)
+
+// Or perhaps this?
+numsBy4 = Array(1...25).map { $0 * 4 }
+print(numsBy4)
+
+// Or perhaps this?
+numsBy4 = Array(stride(from:2, through: 100, by:4))
+print(numsBy4)
+
+
+//You can set the values of the arrays above using whatever method you find the easiest. In previous modules, you were introduced to ranges and sequences in Swift. Leveraging those in the Array initializer will allow you to create the requested arrays in a single line. Don’t let the last two break your stride.
+
+//- Create a function called evenNumbersArray that takes a parameter of [Int] (array of Int) and returns [Int]. The array of Int returned should contain all the even numbers in the array passed. Call the function passing the nums array and print the output.
+
+func evenNumbersArray(_ array: [Int]) -> [Int] {
+  array.filter({ $0 % 2 == 0})
+}
+
+let evenNums = evenNumbersArray(nums)
+print("evenNums \(evenNums)")
+
+//- Create a function called sumOfArray that takes a parameter of [Int?] and returns an Int. The function should return the sum of the array values passed that are not nil. Call the function passing the numsWithNil array, and print out the results.
+
+func sumOfArray(_ array: [Int?]) -> Int {
+  array.compactMap({ $0 }).reduce(0, { $0 + $1 })
```

Comment:
You get the right answer, but I'd normally expect `reduce` to look like this:
````
  array.compactMap({ $0 }).reduce(0, +)
````